lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -174,7 +174,7 @@ def add_creditcard(post, payment, options)
       end
 
       def send_cvv_value?(payment, options)
-        payment.verification_value.nil? || payment.verification_value&.empty? || (options[:stored_credential] && options[:stored_credential][:reason_type] == 'recurring')
+        (payment.verification_value.nil? || payment.verification_value&.empty?) && (options[:stored_credential] && options[:stored_credential][:reason_type] == 'recurring')
```

Comment:
Rethinking this because this condition reverts us back to a state where we send an empty `cvv`. I actually think each of these conditions should be validated independently, so maybe something like this:

We might not even need to check if the reason type is `recurring` here. Thoughts?

Reply:
I actually think this method can just validate the first half of the condition (cvv is not nil and not empty) and then add an `||` condition on line 172 that also checks if the `options[:network_transaction_id]` exists, in which case we also don't send `cvv`.

So, we don't want to rely on `recurring` to tell us to not send `cvv`, since the initial txn can be `recurring` and contain a `cvv` that we need to send because there is no `network_transaction_id`.

Reply:
oh I can see, your changes make sense